code/go/0chain.net/smartcontract/storagesc/sc.go

```diff
@@ -113,7 +113,7 @@ func (ssc *StorageSmartContract) setSC(sc *sci.SmartContract, bcContext sci.BCCo
 	// challenge pool
 	ssc.SmartContract.RestHandlers["/getChallengePoolStat"] = ssc.getChallengePoolStatHandler
 	// events db
-	ssc.SmartContractExecutionStats["/get_transaction"] = ssc.GetTransactionByHashHandler
+	ssc.SmartContract.RestHandlers["/get_transaction"] = ssc.GetTransactionByHashHandler
```

Comment:
Add a new API in addition to the which queries the transaction by Hash, and add one api `/transaction/search` where I can provide multiple query params as per my need.
This is in conjunction to what explorer uses as of now 
```
/transaction/:hash: To query a transaction by its hash
/transactions/search: API to search transactions by providing multiple query params
/transactions/block-hash: API to get transactions by a block's hash.
```

Reply:
1. `/transaction/:hash:` is not possible because we are using map (I hate this approach) as key to the handler which makes it impossible for us to implement URL parameters naturally.
2. Why do we need two APIs (`/transactions/search`, `/transactions/block-hash`) doing the same thing. Just for specialization.

Reply:
I don't understand the point here?
You have two endpoints
```go

ssc.SmartContractExecutionStats["/get_transaction_by_hash"] = ssc.GetTransactionByHashHandler
ssc.SmartContractExecutionStats["/get_transaction_by_client"] = ssc.GetTransactionByClientHandler
```
Where is the problem?

Reply:
I agree with @samtholiya that having two endpoints would be more semantic:

a. `/transaction/:id` - single txn
b. `/transactions` - multiple txns, searched by query params

To reiterate, now we use `client_id` and `metadata` (or maybe we could use just filename? which is our current use case).

Reply:
@shaktals unfortunately a. is not possible due to map implementation so we would be having
a. `/get_transaction_by_hash?hash=:id`
b. `/get_transaction_by_client?client=:id`
c. `/get_transaaction_by_metadata?metadata=:id`
and so on...
cc: @kushthedude

Reply:
Really? That looks real ugly :sweat_smile:

Reply:
Here is a [nice article](https://www.freecodecamp.org/news/rest-api-best-practices-rest-endpoint-design-examples/) about semantic REST endpoints

Reply:
@Sriep @shaktals @kushthedude finalize what you guys want... and I will change that signature accordingly 😅

Reply:
For me, both your last suggestions are ok.
The last one still being the cleanest.
If it's not too much work to remove the `map` implementation, I say go for that.

Reply:
For me, both options you mention are ok.
The last one still being the cleanest.
If it's not too much work to remove the map implementation, I say go for that.